lib/cdo/delete_accounts_helper.rb

```diff
@@ -281,10 +290,25 @@ def assert_constraint(condition, message)
   end
 
   def purge_user_authentications(user)
+    @log.puts "Deleting user authentication options"
     # Delete most recently destroyed (soft-deleted) record first
     user.authentication_options.with_deleted.order(deleted_at: :desc).each(&:really_destroy!)
   end
 
+  def purge_contact_rollups(email)
+    @log.puts "Deleting ContactRollups records for email #{email}"
+    return unless email
+
+    # Contact rollups data are in 4 tables: ContactRollupsRaw, ContactRollupsProcessed,
+    # ContactRollupsPardotMemory, and ContactRollupsFinal.
+    # During daily contact rollups runs, ContactRollupsRaw and ContactRollupsProcessed
+    # are dropped, records marked for deletion in ContactRollupsPardotMemory are
+    # also purged. In addition, records in Pardot server are also deleted.
+    # Thus, only need to delete ContactRollupsFinal record here.
+    ContactRollupsFinal.find_by_email(email)&.destroy
```

Comment:
Do you need to call `really_destroy`? Or set the email field to `nil`?

Reply:
ContactRollupsFinal does not use paranoia features. `destroy` will delete records from database (after triggering model hooks).